src/main/java/com/pileproject/drivecommand/model/nxt/NxtMachineSimulator.java

```diff
@@ -0,0 +1,52 @@
+package com.pileproject.drivecommand.model.nxt;
+
+import com.pileproject.drivecommand.machine.device.DeviceType;
+import com.pileproject.drivecommand.machine.device.input.LineSensor;
+import com.pileproject.drivecommand.machine.device.input.SoundSensor;
+import com.pileproject.drivecommand.machine.device.input.TouchSensor;
+import com.pileproject.drivecommand.machine.device.input.simulation.SimulatedLineSensor;
+import com.pileproject.drivecommand.machine.device.input.simulation.SimulatedSoundSensor;
+import com.pileproject.drivecommand.machine.device.input.simulation.SimulatedTouchSensor;
+import com.pileproject.drivecommand.machine.device.output.Motor;
+import com.pileproject.drivecommand.machine.device.output.simulation.SimulatedMotor;
+import com.pileproject.drivecommand.machine.device.port.InputPort;
+import com.pileproject.drivecommand.machine.device.port.OutputPort;
+
+public class NxtMachineSimulator extends NxtMachine {
```

Comment:
Okay... let me summarize your comments.

### How about making a common Mock class

Yeah, it's a reasonable way. But, what use cases do you imagine? I think we should refer some use cases (or a purpose) for each case (a common Mock class / Mock classes derived from `*Machine`s).

I feel "Mock classes derived from `*Machine`s" is a case that we, [drive](https://github.com/PileProject/drive) developers, come up with at first because we have separated apps and these classes will be used in the corresponding apps which require a concrete machine class as an argument for their controllers (these controllers have some naming problems as @myusak mentioned. please see the next section).

### The name, `NxtController`

Some problems of the name; 
- It doesn't mention about the form of NXT but, in [drive](https://github.com/PileProject/drive), the class specifies the form/sensors/motors as @amiq11 mentioned (these controllers specify the forms). So the name is too general for the purpose from a viewpoint Maybe, we should rename it like `NxtCarController` if we introduce it in this library.
- It indeed says `this class controls an NXT`. So, the class should take a `NxtMachine` or one of its children. This is a case in which we need such mocking classes, not a general mocking class.


### In short...
I think this is a very complicated problem and you both are right. It depends on situations. 


### What we should discuss

- What cases do you suppose with the 2 options? (i.e., what is the purpose of each option?)
- Should we choose one of them? Or can we choose both?
- How much should we do in this library?
  - Assuming all cases is impossible
- Are some design changes necessary ? (In particular, as for arguments etc.)